stdlib/public/core/Collection.swift

```diff
@@ -1199,18 +1199,18 @@ extension Collection {
       return []
     }
 
-    var result = ContiguousArray<T>()
-    result.reserveCapacity(n)
-
-    var i = self.startIndex
-
-    for _ in 0..<n {
-      result.append(try transform(self[i]))
-      formIndex(after: &i)
+    return try unsafe Array<T>(unsafeUninitializedCapacity: n) { (
+      storage: inout UnsafeMutableBufferPointer<T>,
+      initializedCount: inout Int
+    ) throws(E) -> Void in
+      var collectionIndex = self.startIndex
+      for bufferIndex in 0 ..< n {
+        unsafe storage[bufferIndex] = try transform(self[collectionIndex])
```

Comment:
Note that `storage` is uninitialized, so it should be this:
```suggestion
        unsafe storage.initializeElement(at: bufferIndex, to: try transform(self[collectionIndex])
```

Reply:
oh good point! also cleaner

Reply:
Oh! I actually wasn't aware that only works for trivial types! Thanks for that fix!